model_compression_toolkit/quantizers_infrastructure/pytorch/quantize_wrapper.py

```diff
@@ -41,14 +42,23 @@ def __init__(self,
             """
             super().__init__()
             self.dispatcher = dispatcher
-            self.add_module('layer', module)
+            if isinstance(module, nn.Module):
+                self.add_module(LAYER, module)
+            else:
+                # Functional layers
+                setattr(self, LAYER, module)
+
             self._weight_vars = []
 
             # Init weights quantizers
             for name, quantizer in self.dispatcher.weight_quantizers.items():
                 weight = getattr(self.layer, name)
+                weight = weight.detach()
+                delattr(self.layer, name)
```

Comment:
Is deleting the attribute necessary before setting it to a new value?

Reply:
Apparently yes